src/Command/RequeueCommand.php

```diff
@@ -0,0 +1,167 @@
+            'help' => 'Requeue jobs by the job class.',
+        ]);
+        $parser->addOption('queue', [
+            'help' => 'Requeue jobs by the queue the job was received on.',
+        ]);
+        $parser->addOption('config', [
+            'help' => 'Requeue jobs by the config used to queue the job.',
+        ]);
+        $parser->addOption('assume-yes', [
+            'help' => 'Automatically assume yes in response to confirmation prompt.',
+            'short' => 'y',
+            'boolean' => true,
+        ]);
+
+        return $parser;
+    }
+
+    /**
+     * @param \Cake\Console\Arguments $args Arguments
+     * @param \Cake\Console\ConsoleIo $io ConsoleIo
+     * @return void
+     */
+    public function execute(Arguments $args, ConsoleIo $io)
+    {
+        /** @var \Cake\Queue\Model\Table\FailedJobsTable $failedJobsTable */
+        $failedJobsTable = $this->getTableLocator()->get('Cake/Queue.FailedJobs');
+
+        $jobsToRequeue = $failedJobsTable->find();
+
+        if ($args->hasArgument('ids')) {
+            $idsArg = $args->getArgument('ids');
+
+            if ($idsArg !== null) {
+                $ids = explode(',', $idsArg);
+
+                $jobsToRequeue->whereInList($failedJobsTable->aliasField('id'), $ids);
+            }
+        }
+
+        if ($args->hasOption('class')) {
+            $jobsToRequeue->where([
+                $failedJobsTable->aliasField('class') => $args->getOption('class'),
+            ]);
+        }
+
+        if ($args->hasOption('queue')) {
+            $jobsToRequeue->where([
+                $failedJobsTable->aliasField('queue') => $args->getOption('queue'),
+            ]);
+        }
+
+        if ($args->hasOption('config')) {
+            $jobsToRequeue->where([
+                $failedJobsTable->aliasField('config') => $args->getOption('config'),
+            ]);
+        }
+
+        $requeueingCount = $jobsToRequeue->count();
+
+        if (!$requeueingCount) {
+            $io->out('0 jobs found.');
+
+            return;
+        }
+
+        if (!$args->getOption('assume-yes')) {
+            $confirmed = $io->askChoice("Requeue {$requeueingCount} jobs?", ['y', 'n'], 'n');
+
+            if ($confirmed !== 'y') {
+                return;
+            }
+        }
+
+        $io->out("Requeueing {$requeueingCount} jobs.");
+
+        $succeededCount = 0;
+        $failedCount = 0;
+
+        foreach ($jobsToRequeue as $failedJob) {
+            $io->verbose("Requeueing FailedJob with ID {$failedJob->id}.");
+            try {
+                QueueManager::push(
+                    [$failedJob->class, $failedJob->method],
+                    $failedJob->decoded_data,
+                    [
+                        'config' => $failedJob->config,
+                        'priority' => $failedJob->priority,
+                        'queue' => $failedJob->queue,
+                    ]
+                );
+
+                $failedJobsTable->deleteOrFail($failedJob);
```

Comment:
You mean because if `deleteOrFail` fails, the job would remain in the `failed_jobs` table and could be requeued, right? I can note that in the docs. I suppose the only solution to that would be to remove the job from the database, queue it, then insert the record again if there was an exception when queueing. But then a failed job could be "dropped" if it could not be inserted again, which seems worse.